Tests/Unit/MSFT_xDefaultGatewayAddress.Tests.ps1

```diff
@@ -50,193 +80,259 @@ InModuleScope MSFT_xDefaultGatewayAddress {
                 $Result.Address | Should BeNullOrEmpty
             }
         }
-
     }
 
-    Describe 'Test-Properties' {
+    #######################################################################################
 
-        Context 'invoking without -Apply switch and default gateway is exists' {
+    Describe 'Set-TargetResource' {
 
-            #region Mocks
-            Mock Get-NetRoute -MockWith {
-                [PSCustomObject]@{
-                    NextHop = '192.168.0.1'
-                    DestinationPrefix = '0.0.0.0/0'
-                    InterfaceAlias = 'Ethernet'
-                    InterfaceIndex = 1
-                    AddressFamily = 'IPv4'
-                }
+        #region Mocks
+        Mock Get-NetAdapter -MockWith { [PSObject]@{ Name = 'Ethernet' } }
+
+        Mock Get-NetRoute -MockWith {
+            [PSCustomObject]@{
+                NextHop = '192.168.0.1'
+                DestinationPrefix = '0.0.0.0/0'
+                InterfaceAlias = 'Ethernet'
+                InterfaceIndex = 1
+                AddressFamily = 'IPv4'
             }
-            #endregion
+        }
+
+        Mock Remove-NetRoute
 
-            It 'when default gateway does not match should be $false' {
+        Mock New-NetRoute
+
+        #endregion
+
+        Context 'invoking with no Default Gateway Address' {
+            It 'should rerturn $null' {
                 $Splat = @{
-                    Address = '10.0.0.2'
                     InterfaceAlias = 'Ethernet'
                     AddressFamily = 'IPv4'
                 }
-                $Result = Test-Properties @Splat
-                $Result | Should Be $false
+                { $Result = Set-TargetResource @Splat } | Should Not Throw
+                $Result | Should BeNullOrEmpty
             }
 
-            It 'when default gateway matches should be $true' {
-                $Splat = @{
-                    Address = '192.168.0.1'
-                    InterfaceAlias = 'Ethernet'
-                    AddressFamily = 'IPv4'
-                }
-                $Result = Test-Properties @Splat
-                $Result | Should Be $true
+            It 'should call all the mocks' {
+                Assert-MockCalled -commandName Get-NetAdapter -Exactly 0
```

Comment:
Good catch! I used to call test-resourceproperty in this function, but removed it. Didn't remove the mock. Have now :)